broker/src/main/java/io/camunda/zeebe/broker/transport/partitionapi/InterPartitionCommandReceiverImpl.java

```diff
@@ -27,6 +30,7 @@ public final class InterPartitionCommandReceiverImpl {
   private final Decoder decoder = new Decoder();
   private final LogStreamRecordWriter logStreamWriter;
   private boolean diskSpaceAvailable = true;
+  private long checkpointId = CheckpointState.NO_CHECKPOINT;
```

Comment:
:thought_balloon: Just remembered on edge case here. After failover, it should get the latest checkpoint. Since no new checkpoint is created, the listener won't be notified. How should we handle this? It probably doesn't have to be solved in this PR. But we should have some idea how we are going to handle it.

Reply:
It might be a problem in the sender as well. :thinking:

Reply:
@deepthidevaki Oh yeah, I thought that registering as `CheckpointListener` should invoke the `onNewCheckpoint` method with the current checkpoint. That should happen before the actor has fully started, so it should be safe?

Reply:
If you looks at the current `CheckpointRecordsProcessor`, it notifies only when a new checkpoint is created. I think this should be handled in the `CheckpointRecordsProcessor`. 

I assume the listener will be registered when the `CheckpointRecordsProcessor` is created. We haven't added that part yet. So it is not clear how it will be. I guess, if the listener is notified when the processor is initialized (reading the zeebe state) and when replay result in new checkpointId it should be fine. I will create a followup issue.

Reply:
Maybe registering is not the right moment to do this, instead we should also notify listeners after `init` and during `replay`. OTOH, **not** notifying the listener is also ok, or at least safe, because at worst a redundant checkpoint create record is written, right?

Reply:
For the receiver it is safe. But the sender should have the correct checkpointId.

Reply:
Right, and that means that the sender can only be used after `init` was called :grimacing:

Reply:
Which is currently the case, because the sender is only used either during regular processing and by `PendingProcessMessageSubscriptionChecker` and `DeploymentRedistributor` which only start once `StreamProcessorLifecycleAware#onRecovered` is called.